youtube_dl/extractor/rai.py

```diff
@@ -238,53 +219,79 @@ class RaiPlayLiveIE(RaiBaseIE):
     }
 
     def _real_extract(self, url):
-        display_id = self._match_id(url)
+        mobj = re.match(self._VALID_URL, url)
+        base, display_id, = mobj.group('base', 'id')
 
-        webpage = self._download_webpage(url, display_id)
+        media = self._download_json(
+            '%s.json' % urljoin(base, 'dirette/' + display_id),
+            display_id, 'Downloading channel JSON')
+
+        title = media['name']
+        video = media['video']
+        video_id = media['id'].replace('ContentItem-', '')
 
-        video_id = self._search_regex(
-            r'data-uniquename=["\']ContentItem-(%s)' % RaiBaseIE._UUID_RE,
-            webpage, 'content id')
+        relinker_info = self._extract_relinker_info(video['content_url'], video_id)
+        self._sort_formats(relinker_info['formats'])
 
-        return {
-            '_type': 'url_transparent',
-            'ie_key': RaiPlayIE.ie_key(),
-            'url': 'http://www.raiplay.it/dirette/ContentItem-%s.html' % video_id,
+        info = {
             'id': video_id,
             'display_id': display_id,
+            'title': self._live_title(title) if relinker_info.get(
+                'is_live') else title,
+            'alt_title': media.get('subtitle'),
+            'description': media.get('description'),
+            'uploader': strip_or_none(media.get('channel')),
+            'creator': strip_or_none(media.get('editor')),
+            'duration': parse_duration(video.get('duration')),
         }
 
+        info.update(relinker_info)
+        return info
+
 
 class RaiPlayPlaylistIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?raiplay\.it/programmi/(?P<id>[^/?#&]+)'
+    _VALID_URL = r'(?P<base>https?://(?:www\.)?raiplay\.it/)programmi/(?P<id>[^/?#&]+)'
```

Comment:
the part you're matching is static and won't change, I meant to match to the end of the URL(the same for `RaiPlayLiveIE`).

Reply:
Sorry but then I don't get it.  Please elaborate further.

Reply:
Do you meant to add possible trailing `/` or possible trailing extension (`.html` and/or `.json` if any)?

Reply:
```suggestion
    _VALID_URL = r'(?P<base>https?://(?:www\.)?raiplay\.it/programmi/(?P<id>[^/?#&]+))'
```

Reply:
Thanks!

This should be addressed by https://github.com/ytdl-org/youtube-dl/pull/27077/commits/65600a1efd2891995a5fb6d96d32c77969b1a3ef